website/server/models/task.js

```diff
@@ -167,6 +182,20 @@ TaskSchema.statics.fromJSONV2 = function fromJSONV2 (taskObj) {
 
 export let Task = mongoose.model('Task', TaskSchema);
 
+Task.schema.path('shortName').validate(function valiateShortNameNotTaken (shortName, respond) {
+  Task.findOne({
+    _id: { $ne: this._id },
+    userId: this.userId,
+    shortName,
+  }).exec().then((task) => {
+    let shortNameAvailable = !task;
+
+    respond(shortNameAvailable);
+  }).catch(() => {
```

Comment:
It's necessary in this case. We want the error to be handled by mongoose not by the global catcher